### DISCUSSION TITLE
if else in automation

### DISCUSSION THREAD

**Discussion by Promphut** [Budibase/budibase#6628] [Ideas]
if else in automation will make an automation very powerful.

**Comment by melohagan** [comment]
Hi @Promphut 

I agree! 

There is a 'Stop If' automation step that can terminate the flow. 
https://docs.budibase.com/docs/conditions-1

Therefore you could have two automations, one for each 'branch' if you need an If/Else. Obviously if you have lots of branching it could get very complicated, but I don't know how you would display a lot of branching in a nice way within the UI.

Also, the automation duplication is yet to be released, but the work is done: https://github.com/Budibase/budibase/pull/6603

**Reply by ghost** [reply]
I second this feature request. There are so many scenarios where I want to implement an if/else condition in my automations, specifically around handling data in the backend if using the BB DBs. 

@melohagan - In terms of UI, there are lots of nice examples of flow-building tools out there. I think Shopify Flows has a a pretty intuitive UI. Below is a screenshow of their older UI, but I think it gives you an idea of what it could look like.
![image](https://user-images.githubusercontent.com/64409787/183670435-3046df23-8091-408e-a7c9-3e34278bd14b.png)

**Reply by R2bEEaton** [reply]
Here's my take on how this could be done in a Budibase-y way #7553

**Reply by GregFdz** [reply]
Kind of unearthing the topic, but...

I think that the main problem of "else" is the UI. It would get really complex. But I think that there are simple solutions. For starters, adding an "if" field to each step block would solve a big chunk of this problem. A goto block or even the possibility to create actions within the js like createRow(table, rowid, [rowname, rowcontent], [rowname, rowcontent]...) would make building a "else" block" much less relevant.

But if someone was to create a really unlimited conditionnal model, he'd probably need to go with a vertical tree structure just like the components in the "design" section. Or folders / pages like github. You creat a new if, you click on it an it displays a page with your conditions, ability to add an else / elseif block and a "previous button". And at the top of the page, the path is diplayed, just like on github and the attached picture : 
![image](https://user-images.githubusercontent.com/25699677/206463444-5d008d92-0ffe-4033-9107-61b63b48f6a1.png)

These solution almost all require something : naming steps. Even now, it's a pain in the... you create an automation and then you realize that you have to add a step ontop of the others... you have to rewrite all the bindings.
